packages/api/src/external/carequality/patient.ts

```diff
@@ -25,41 +29,45 @@ import {
   getOutboundPatientDiscoveryRespCount,
   getOutboundPatientDiscoveryResps,
 } from "./command/outbound-patient-discovery-resp/get-outbound-patient-discovery-resp";
-import { createPatientDiscoveryRequest } from "./create-pd-request";
 import { CQLink } from "./cq-patient-data";
-import { OutboundPatientDiscoveryResp } from "./patient-discovery-result";
+import { createPatientDiscoveryRequest } from "./create-pd-request";
 import { cqOrgsToXCPDGateways, generateIdsForGateways } from "./organization-conversion";
-import { MedicalDataSource } from "@metriport/core/external/index";
+import { OutboundPatientDiscoveryResp } from "./patient-discovery-result";
 import { PatientDataCarequality } from "./patient-shared";
-import { getCQGateways } from "./command/cq-directory/cq-gateways";
 
 dayjs.extend(duration);
 
+const context = "cq.patient.discover";
+const iheGateway = makeIheGatewayAPIForPatientDiscovery();
+
+export const PATIENT_DISCOVERY_TIMEOUT = dayjs.duration({ minutes: 0.25 });
+const CHECK_DB_INTERVAL = dayjs.duration({ seconds: 5 });
+
 export function getCQData(
   data: PatientExternalData | undefined
 ): PatientDataCarequality | undefined {
   if (!data) return undefined;
   return data[MedicalDataSource.CAREQUALITY] as PatientDataCarequality; // TODO validate the type
 }
 
-const createContext = "cq.patient.discover";
+export async function discover(patient: Patient, facilityNPI: string): Promise<void> {
+  const baseLogMessage = `CQ PD - M patientId ${patient.id}`;
+  const { log: outerLog } = out(baseLogMessage);
+  const { cxId } = patient;
 
-export const PATIENT_DISCOVERY_TIMEOUT = dayjs.duration({ minutes: 0.25 });
-const CHECK_DB_INTERVAL = dayjs.duration({ seconds: 5 });
+  if (!iheGateway) return outerLog(`IHE GW not available, skipping PD`);
+  if (!(await isCQDirectEnabledForCx(cxId))) {
+    return outerLog(`CQ disabled for cx ${cxId}, skipping PD`);
+  }
 
-export async function discover(patient: Patient, facilityNPI: string): Promise<void> {
-  const { log } = Util.out(`CQ discover - M patientId ${patient.id}`);
   try {
-    const iheGateway = makeIheGatewayAPI();
-    if (!iheGateway) return;
-
-    const { cxId } = patient;
     const pdRequest = await prepareForPatientDiscovery(patient, facilityNPI);
     const numGateways = pdRequest.gateways.length;
 
-    log(`Kicking off patient discovery. RequestID: ${pdRequest.id}`);
-    // Intentionally asynchronous - we will be checking for the results in the database
-    iheGateway.startPatientDiscovery(pdRequest);
+    const { log } = out(`${baseLogMessage}, requestId: ${pdRequest.id}`);
+
+    log(`Kicking off patient discovery`);
+    await iheGateway.startPatientDiscovery(pdRequest);
```

Comment:
Important change, we were not awaiting nor doing error handling if it failed.

Since this is an asynchronous integration (IHE GW accepts the request and makes a new request when done), its safe to await.

Reply:
agreed we should always be awaiting IHE GW invocations - they're processed async, but we need to confirm that they were delivered

should be the same for DQ, DR, etc